### PR TITLE
Cleanup GraphicsDevice

### PR DESCRIPTION
**Pull request by cra0zy**

### CODE REVIEW

**Comment by Jjagg** [comment]
Thanks @cra0zy!
This is definitely cleaner :)

_Sent from my Motorola Moto G (5) using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

File: MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs [hunk 1]
```diff
@@ -567,7 +570,15 @@ internal void OnPresentationChanged()
...
-        partial void PlatformValidatePresentationParameters(PresentationParameters presentationParameters)
+        partial void PlatformReset()
+        {
+#if WINDOWS
+            CorrectBackBufferSize();
+#endif
+            PlatformValidatePresentationParameters(PresentationParameters);
+        }
+
+        private void PlatformValidatePresentationParameters(PresentationParameters presentationParameters)
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs]: Looks like no one else calls `PlatformValidatePresentationParameters` anymore... should I just get merged with `PlatformReset`?

Or did I miss somewhere `PlatformValidatePresentationParameters` is used?

Comment (Jjagg) [reply MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs]: Yeah, it can be merged.

Comment (cra0zy) [reply MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs]: Done.

File: MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs [hunk 2]
```diff
@@ -567,16 +570,20 @@ internal void OnPresentationChanged()
...
+            CorrectBackBufferSize();
+#endif
+
 #if WINDOWS_UAP
-            if (presentationParameters.SwapChainPanel == null)
+            if (PresentationParameters.SwapChainPanel == null)
                 throw new ArgumentException("PresentationParameters.SwapChainPanel must not be null.");
 #elif WINDOWS_STOREAPP
-            if (presentationParameters.DeviceWindowHandle == IntPtr.Zero && presentationParameters.SwapChainBackgroundPanel == null)
+            if (PresentationParameters.DeviceWindowHandle == IntPtr.Zero && presentationParameters.SwapChainBackgroundPanel == null)
```
Comment (Jjagg) [inline MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs]: @cra0zy Missed the lower case `presentationParameters` here

Comment (cra0zy) [reply MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs]: ups.

**Comment by cra0zy** [comment]
@tomspilman All is good now.

**Comment by tomspilman** [comment]
Thanks @cra0zy !

**Pull request merged**